extensions/jackson/runtime/pom.xml

```diff
@@ -33,6 +33,11 @@
             <groupId>io.quarkus</groupId>
             <artifactId>quarkus-arc</artifactId>
         </dependency>
+        <dependency>
+            <groupId>io.vertx</groupId>
+            <artifactId>vertx-core</artifactId>
```

Comment:
Wondering about native compilation, actually. As it would not contain the vert.x substitutions and instructions. 
Another aspect is a CLI app that does not depend on Vert.x would fail, right? Maybe it should be optional and we should to a classloading check before using our pool.

Reply:
This is only done in order to get the extension compling. 
The pool is never used if the Quarkus Vertx  extension is not used

Reply:
I'm not sure that I'm understanding this: if vert.x won't be found on the classpath the jackson's `ObjectMapper` will use the default pool provided by jackson out-of-the-box, that I guess it is what we want? Am I missing something?

Reply:
That is correct

Reply:
FWIW, this pattern is used extensively in Micrometer and OpenTelemetry

Reply:
It seems like we are actually using `<optional>true</optional>` instead of `<scope>provided</scope>` in other extensions, so let's go with that here too

Reply:
Fixed.